src/misc/programs.cpp

```diff
@@ -80,7 +80,7 @@ void PROGRAMS_MakeFile(const char *name, PROGRAMS_Main *main)
 
 	// Register the program's main pointer
 	// NOTE: This step must come after the index is saved in the COM data
-	internal_progs.push_back(main);
+	internal_progs.push_back(creator);
```

Comment:
`emplace_back` (we should get 100% pass-through for the temporary `creator` passed into the function, constructed in-place in the vector)